src/index.ts

```diff
@@ -148,30 +190,77 @@ export class CSSStyleObserver {
   }
 
   /**
-   * Collect CSS variable values and invoke callback.
+   * Process observed variables and collect changes
+   *
+   * @param computedStyle The computed style of the target element
+   * @param processChange Function to process each changed property
+   * @param finalizeChanges Function to finalize and handle collected changes
    */
-  private _handleUpdate(): void {
-    if (this._targetElement) {
-      const computedStyle = getComputedStyle(this._targetElement);
-
-      const changedProperties: CSSDeclarations = {};
-
-      this._observedVariables.forEach(propertyName => {
+  private _processChanges(
```

Comment:
This name doesn’t reflect what it’s doing. It’s more `_processObservedVariables`, no?

Reply:
Renamed `_processChanges` to `_processObservedVariables` to better reflect its functionality